backend/bamboo/database/models.py

```diff
@@ -96,14 +96,22 @@ class Site(Base):
     deploy_target: so.Mapped[Optional[str]]
     deploy_method: so.Mapped[Optional[str]]
     deploy_secret: so.Mapped[Optional[str]] = so.mapped_column(sa.Text)
-    pages: so.WriteOnlyMapped["Page"] = so.relationship(back_populates="site")
-    notifications: so.WriteOnlyMapped["Notification"] = so.relationship(back_populates="site")
-    volunteer_forms: so.WriteOnlyMapped["VolunteerForm"] = so.relationship(back_populates="site")
-    sponsor_forms: so.WriteOnlyMapped["SponsorForm"] = so.relationship(back_populates="site")
-    speaker_forms: so.WriteOnlyMapped["SpeakerForm"] = so.relationship(back_populates="site")
-    talks: so.WriteOnlyMapped["Talk"] = so.relationship(back_populates="site")
-    blogs: so.WriteOnlyMapped["Blog"] = so.relationship(back_populates="site")
-    cities: so.WriteOnlyMapped["City"] = so.relationship(back_populates="site")
+    pages: so.Mapped["Page"] = so.relationship(back_populates="site", passive_deletes=True)
```

Comment:
This will make it return a single `Page` which is not correct.

Change it to `so.DynamicMapped` if you want to read a *lazy* set of pages.

Reply:
> This will make it return a single `Page` which is not correct.
> 
> Change it to `so.DynamicMapped` if you want to read a _lazy_ set of pages.

I got it, but when using `so.DynamicMapped` or `so.WriteOnlyMapped`（lazy loader），you have to set `passive_deletes=True` or it will raise bellow exception when deleting a site:
```
sqlalchemy.exc.InvalidRequestError: Attribute Site.pages can't load the existing state from the database for this operation; full iteration is not permitted.  If this is a delete operation, configure passive_deletes=True on the Site.pages relationship in order to resolve this error.
```
According to SQLAlchemy's [doc](https://github.com/sqlalchemy/sqlalchemy/blob/main/lib/sqlalchemy/orm/_orm_constructors.py#L1464) once you set `passive_delets=True`, it usually implies an ON DELETE <CASCADE|SET NULL> rule, which I think not the right cascade level here. I gonna keep `Mapped` and change type hint to `Mapped[list[page]]`.

Reply:
> I got it, but when using `so.DynamicMapped` or `so.WriteOnlyMapped`（lazy loader），you have to set `passive_deletes=True` or it will raise bellow exception when deleting a site:

it only happens when using `WriteOnlyMapped`, pull the latest changes from master and I've added some deletion cases.

Reply:
Keep WriteOnlyMapped, I'll update my PR to pass unit tests.

Reply:
Updated!